### PR TITLE
[FFI] Add support for string args

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description
Add support for string args

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by jermainewang** [comment]
Does python side already support string args? Also, please double check cython.

**Comment by VoVAllen** [comment]
Yes. 
python: https://github.com/dmlc/dgl/blob/master/python/dgl/_ffi/_ctypes/function.py#L118
cython: https://github.com/dmlc/dgl/blob/master/python/dgl/_ffi/_cython/function.pxi#L132

File: include/dgl/runtime/packed_func.h [hunk 1]
```diff
@@ -398,6 +398,10 @@ class DGLPODValue_ {
     DGL_CHECK_TYPE_CODE(type_code_, kDLInt);
     return value_.v_int64 != 0;
   }
+  operator std::string() const {
+    DGL_CHECK_TYPE_CODE(type_code_, kStr);
+    return std::string(value_.v_str);
+  }
```
Comment (jermainewang) [inline include/dgl/runtime/packed_func.h]: I double checked `DGLArgValue` and `DGLRetValue`. They all have corresponding string support (https://github.com/dmlc/dgl/blob/master/include/dgl/runtime/packed_func.h#L489 and https://github.com/dmlc/dgl/blob/master/include/dgl/runtime/packed_func.h#L578). Why we need this in the base class `DGLPODValue_`? `DGLPODValue_` shouldn't be used directly.

Comment (VoVAllen) [reply include/dgl/runtime/packed_func.h]: You are right. I misunderstood the way of using this part.

**Pull request closed without merging**